docs/DeveloperGuide.adoc

```diff
@@ -98,56 +112,80 @@ image::LogicClassDiagram.png[]
+* a new `CommandAllocator` object is created to allocate the user input to appropriate entity-specific `Parser`. The `CommandAllocator` 's `allocate` method then returns a new Command object which is executed by the LogicManager, or
+* the appropriate `Parser` is directly called if no specifying is required and returns a new  `Command` object which is executed by the `LogicManager`.
+.  The command execution can affect the `Model` (e.g. adding a participant or deleting a team).
 .  The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
 .  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
 
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete participant P-1")` API call.
 
 .Interactions Inside the Logic Component for the `delete 1` Command
 image::DeleteSequenceDiagram.png[]
 
-NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+NOTE: The lifeline for `DeleteParticipantCommandParser` and `DeleteCommandAllocator` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
-image::ModelClassDiagram.png[]
+image::BetterModelClassDiagram.png[]
 
 *API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
 
 The `Model`,
 
 * stores a `UserPref` object that represents the user's preferences.
-* stores the Address Book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
-* does not depend on any of the other three components.
+* stores the lists of our various entities.
+* Model is the bridge between Logic and Storage and provides an abstraction of how the data is stored in memory.
+* It exposes a `ReadableEntityList` which only has the list method to remind Logic that the data given should not be modified.
+* The UI can be bound to these lists so that it automatically updates when the contents of the list change.
+* At the heart of the model are observable lists which allow for the dynamic updating of the UI.
+* The `Model` interface also serves as an API through which controller can edit the data stored in memory.
+
+`ModelManager`
+* ModelManager implements all the methods exposed by the Model Interface. At its heart, it only contains 3 `EntityLists`
+, 3 `FilteredLists` and a `UserPrefs` Object and does all the validation logic needed for the application
+
+.Simple Illustration of ModelManager
+image::ModelManagerClassDiagram.png[]
```

Comment:
The Class Diagrams here are very simple and neat, which makes them easy to understand.